File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/Product.java
package org.morecup.jimmerddd.java.sample.domain.order;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.ManyToMany;
import org.morecup.jimmerddd.java.sample.domain.base.BaseEntity;
import java.math.BigDecimal;
import java.util.List;

@Entity
public interface Product extends BaseEntity {
    String name();
    BigDecimal originalPrice();

    @ManyToMany(mappedBy = "productList")
    List<Order> orderList();
}

File: jimmer-ddd-kotlin-sample/domain/src/main/kotlin/org/morecup/jimmerddd/kotlin/sample/domain/order/Product.kt
package org.morecup.jimmerddd.kotlin.sample.domain.order

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.ManyToMany
import org.morecup.jimmerddd.kotlin.sample.domain.base.BaseEntity
import java.math.BigDecimal

@Entity
interface Product: BaseEntity {
    val name: String
    val originalPrice: BigDecimal

    @ManyToMany(mappedBy = "productList")
    val orderList: List<Order>
}